### PR TITLE
FIX: replace button with dolGetButtonAction on Societe card

### PR DESCRIPTION
**Pull request by atm-GregM**
Need PR #20145 

replace button without dolGetButtonAction on Societe card

before 
image

with fix 
image

### CODE REVIEW

File: htdocs/societe/card.php [hunk 1]
```diff
@@ -2444,10 +2444,20 @@ function init_supplier_categ() {
...
-			$formconfirm = $form->formconfirm($_SERVER["PHP_SELF"]."?socid=".$object->id, $langs->trans("DeleteACompany"), $langs->trans("ConfirmDeleteCompany"), "confirm_delete", '', 0, "action-delete");
+		if ($action == 'delete') {
+			$formquestion = array(
+				array(
+					'label' => $langs->trans('M456465645y'),
+					'value' => 'yes',
+				)
+			);
+
+			$formconfirm .= $form->formconfirm($_SERVER["PHP_SELF"].'?id='.$object->id, $langs->trans('DeleteThirdParty'), $langs->trans('ConfirmDeleteCompany'), 'confirm_delete', '', 0, 2);
```
Comment (atm-john) [inline htdocs/societe/card.php:2455]: Why DeleteACompany become DeleteThirdParty ?

File: htdocs/societe/card.php [hunk 2]
```diff
@@ -2444,10 +2444,20 @@ function init_supplier_categ() {
 		$formconfirm = '';
 
 		// Confirm delete third party
-		if ($action == 'delete' || ($conf->use_javascript_ajax && empty($conf->dol_use_jmobile))) {
-			$formconfirm = $form->formconfirm($_SERVER["PHP_SELF"]."?socid=".$object->id, $langs->trans("DeleteACompany"), $langs->trans("ConfirmDeleteCompany"), "confirm_delete", '', 0, "action-delete");
+		if ($action == 'delete') {
+			$formquestion = array(
+				array(
+					'label' => $langs->trans('M456465645y'),
```
Comment (atm-john) [inline htdocs/societe/card.php:2450]: Why ?

File: htdocs/societe/card.php [hunk 3]
```diff
@@ -2910,15 +2920,21 @@ function init_supplier_categ() {
 				if (empty($user->socid)) {
 					if (!empty($object->email) || $at_least_one_email_contact) {
 						$langs->load("mails");
-						print '<a class="butAction" href="'.$_SERVER['PHP_SELF'].'?socid='.$object->id.'&action=presend&mode=init#formmailbeforetitle">'.$langs->trans('SendMail').'</a>'."\n";
+						$params = array(
+							'attr' => array(
+								'title' => ''
+							)
+						);
```
Comment (atm-john) [inline htdocs/societe/card.php:2927]: If $param is empty don't set it and leave default value

File: htdocs/societe/card.php [hunk 4]
```diff
@@ -2910,15 +2920,21 @@ function init_supplier_categ() {
...
 						$langs->load("mails");
-						print '<a class="butActionRefused classfortooltip" href="#" title="'.dol_escape_htmltag($langs->trans("NoEMail")).'">'.$langs->trans('SendMail').'</a>'."\n";
+						$params['attr']['title'] = $langs->trans('NoEMail');
+						print dolGetButtonAction($langs->trans('SendMail'), '', 'default', $_SERVER['PHP_SELF'].'?socid='.$object->id.'&action=presend&mode=init#formmailbeforetitle'.newToken(), '', false, $params);
 					}
 				}
 
 				if ($user->rights->societe->creer) {
-					print '<a class="butAction" href="'.$_SERVER["PHP_SELF"].'?socid='.$object->id.'&action=edit&token='.newToken().'">'.$langs->trans("Modify").'</a>'."\n";
+					print dolGetButtonAction($langs->trans('Modify'), '', 'default', $_SERVER["PHP_SELF"].'?socid='.$object->id.'&action=edit&token='.newToken(), '', $permissiontoadd);
```
Comment (atm-john) [inline htdocs/societe/card.php:2937]: remplacer $permissiontoadd par $user->rights->societe->creer a moins que $permissiontoadd soit egale à $user->rights->societe->creer

Comment (atm-john) [reply htdocs/societe/card.php:2937]: si tel est le cas tu peux enlever la condition 
```<?php 
if ($user->rights->societe->creer) {
```

File: htdocs/societe/card.php [hunk 5]
```diff
@@ -2930,16 +2946,9 @@ function init_supplier_categ() {
...
-
-				if ($user->rights->societe->supprimer) {
-					if ($conf->use_javascript_ajax && empty($conf->dol_use_jmobile)) {	// We can't use preloaded confirm form with jmobile
-						print '<span id="action-delete" class="butActionDelete">'.$langs->trans('Delete').'</span>'."\n";
-					} else {
-						print '<a class="butActionDelete" href="'.$_SERVER["PHP_SELF"].'?socid='.$object->id.'&action=delete&token='.newToken().'">'.$langs->trans('Delete').'</a>'."\n";
-					}
+					print dolGetButtonAction($langs->trans('MergeThirdparties'), $langs->trans('Merge'), 'danger', $_SERVER["PHP_SELF"].'?socid='.$object->id.'&action=merge&token='.newToken(), '', $permissiontoadd);
 				}
+				print dolGetButtonAction('', $langs->trans('Delete'), 'delete', $_SERVER["PHP_SELF"].'?socid='.$object->id.'&action=delete&token='.newToken(), '', $user->rights->societe->supprimer);
```
Comment (atm-john) [inline htdocs/societe/card.php:2951]: reprend cette partie là fait perdre une fonctionnalité

**Review by john-botella — CHANGES_REQUESTED** [review]

File: htdocs/societe/card.php (hunk 1 above)
Comment (atm-GregM) [reply htdocs/societe/card.php:2455]: replace to "DeleteACompany"
DeleteThirdParty doesn't exist in companies.lang

File: htdocs/societe/card.php (hunk 2 above)
Comment (atm-GregM) [reply htdocs/societe/card.php:2450]: deleted

File: htdocs/societe/card.php (hunk 3 above)
Comment (atm-GregM) [reply htdocs/societe/card.php:2927]: add on function dolGetButtonAction

File: htdocs/societe/card.php (hunk 4 above)
Comment (atm-GregM) [reply htdocs/societe/card.php:2937]: image

deleted condition

**Pull request merged**